addons/account_peppol/static/src/components/peppol_radio_field/peppol_radio_field.js

```diff
@@ -0,0 +1,44 @@
+/** @odoo-module **/
+
+import { evaluateExpr } from "@web/core/py_js/py";
+import { RadioField, radioField } from "@web/views/fields/radio/radio_field";
+import { registry } from "@web/core/registry";
+
+class PeppolRadioField extends RadioField {
+    static template = "account_peppol.PeppolRadioField";
+    static props = {
+        ...RadioField.props,
+        hiddenItems: { type: String, optional: true },
+        readonlyItems: { type: String, optional: true },
+    };
+
+    setup() {
+        super.setup()
+        this.initialSelection = this.props.record.data[this.props.name];
+    }
+
+    get hiddenItems() {
+        return this.props.hiddenItems ? evaluateExpr(this.props.hiddenItems, this.props.record.evalContext).filter(item => (item != this.initialSelection)) : [];
```

Comment:
You are evaluating multiple time the input props. Evaluate them once in the setup instead
```suggestion
    setup() {
        super.setup()
        this.hiddenItems = evaluateExpr(this.props.hiddenItems || "[]", this.props.record.evalContext);
    }
```
I'm not sure about your `item != this.initialSelection`. Are you not comparing an item with a list here?

Reply:
this resolves to a string
```js
this.initialSelction = this.props.record.data[this.props.name]
```
![image](https://github.com/odoo/odoo/assets/46533567/b747ecb1-885a-4ad7-af4f-c7d1ab9c4d50)

You're definitely right about evaluating the expressions in the setup though :+1: